internal/installercache/installercache.go

```diff
@@ -177,41 +268,49 @@ func (i *Installers) evict() {
 			return nil
 		}
 
-		//save the other files based on their mod time
-		files.Add(&fileInfo{path, info})
-		totalSize += info.Size()
+		includeFile := false
+		stat, ok := info.Sys().(*syscall.Stat_t)
+		if !ok {
+			i.log.Warnf("could not determine hardlink status for %s - item will not be filtered", path)
+			includeFile = true
+		}
+		if stat.Nlink == 1 { // This means we have no hardlinks to the file.
+			includeFile = true
+		}
+
+		// if the file has no hardlinks then add it to the queue
+		if includeFile {
+			files.Add(&fileInfo{path, info})
+			totalSize += info.Size()
+		}
```

Comment:
This intermediate variable is not needed. Something like this would work

```suggestion
		stat, ok := info.Sys().(*syscall.Stat_t)
		if !ok || stat.Nlink == 1 {
		        if !ok {
  			      i.log.Warnf("could not determine hardlink status for %s - item will not be filtered", path)
  			}
			files.Add(&fileInfo{path, info})
			totalSize += info.Size()
		}
```